gui/src/app/pages/HomePage/HomePage.tsx

```diff
@@ -128,79 +111,95 @@ const determineShouldBeInitiallyCollapsed = (width: number) => {
             height={height}
-            initialPosition={height / 3}
+            initialPosition={2 * height / 3}
         >
-            <DataFileEditor
+            <StanFileEditor
                 width={0}
                 height={0}
-                fileName="data.json"
-                fileContent={dataFileContent}
-                onSaveContent={saveDataFileContent}
-                editedFileContent={editedDataFileContent}
-                setEditedFileContent={setEditedDataFileContent}
+                fileName="main.stan"
+                fileContent={data.stanFileContent}
+                // this could be made more ergonomic?
+                onSaveContent={() => update({ type: 'commitFile', filename: SPAnalysisKnownFiles.STANFILE })}
+                editedFileContent={data.ephemera.stanFileContent}
+                setEditedFileContent={(content: string) => update({ type: 'editFile', content, filename: SPAnalysisKnownFiles.STANFILE })}
                 readOnly={false}
+                setCompiledUrl={setCompiledMainJsUrl}
             />
-            <LowerRightView
+            <DataFileEditor
                 width={0}
                 height={0}
-                compiledMainJsUrl={compiledMainJsUrl}
-                dataFileContent={dataFileContent}
-                dataIsSaved={dataFileContent === editedDataFileContent}
-                samplingOpts={samplingOpts}
-                setSamplingOpts={setSamplingOpts}
+                fileName="data.json"
+                fileContent={data.dataFileContent}
+                onSaveContent={() => update({ type: 'commitFile', filename: SPAnalysisKnownFiles.DATAFILE })}
+                editedFileContent={data.ephemera.dataFileContent}
+                setEditedFileContent={(content: string) => update({ type: 'editFile', content, filename: SPAnalysisKnownFiles.DATAFILE })}
+                readOnly={false}
             />
         </Splitter>
     )
 }
 
+type RightViewProps = {
+    width: number
+    height: number
+    compiledMainJsUrl?: string
+}
+
+const RightView: FunctionComponent<RightViewProps> = ({ width, height, compiledMainJsUrl }) => {
+    return (
+        <LowerRightView
+            width={width}
+            height={height}
+            compiledMainJsUrl={compiledMainJsUrl}
+        />
+    )
+}
+
 type LowerRightViewProps = {
     width: number
     height: number
     compiledMainJsUrl?: string
-    dataFileContent: string
-    dataIsSaved: boolean
-    samplingOpts: SamplingOpts
-    setSamplingOpts: (opts: SamplingOpts) => void
 }
 
-const LowerRightView: FunctionComponent<LowerRightViewProps> = ({ width, height, compiledMainJsUrl, dataFileContent, dataIsSaved, samplingOpts, setSamplingOpts }) => {
+const LowerRightView: FunctionComponent<LowerRightViewProps> = ({ width, height, compiledMainJsUrl }) => {
+    const { data, update } = useContext(SPAnalysisContext)
+    const dataIsSaved = data.dataFileContent === data.ephemera.dataFileContent
```

Comment:
I see this as a definition internal to the data model, & think it'd probably be better to keep this internal to that file & expose the check via a function (though that function would probably just be `const dataIsSaved = (data: SPAnalysisDataModel) => (data.dataFileContent === data.ephemera.dataFileContent)`)

Reply:
I have added the function and have named it modelHasUnsavedDataFileChanges to go along with the existing modelHasUnsavedChanges